runtime/bcutil/cfreader.c

```diff
@@ -2442,8 +2449,13 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 		classfile->accessFlags |= CFR_ACC_ABSTRACT;
 	}
 
+	/* Java 9 Spec: If the ACC_MODULE flag is set in the access_flags item, then no
+	 * other flag in the access_flags item should be set. If other flags are set, then this
+	 * file is not a module.
+	 */
 	if ((vmVersionShifted >= BCT_Java9MajorVersionShifted)
-			&& (J9_ARE_ALL_BITS_SET(classfile->accessFlags, CFR_ACC_MODULE))) {
+			&& (classfile->majorVersion >= 53)
```

Comment:
Is this check necessary after checking vmVersionShifted?

Reply:
These are not both needed.